R/helpers.R

```diff
@@ -1758,3 +1767,89 @@ gt_latex_dependencies <- function() {
          call. = FALSE)
   }
 }
+
+#' A vector of default fonts for use with **gt** tables
+#'
+#' The vector of fonts given by `default_fonts()` should be used with a **gt**
+#' table that is rendered to HTML. We can specify additional fonts to use but
+#' this default set should be placed after that to act as fallbacks. This is
+#' useful when specifying `font` values in the [cell_text()] function (itself
+#' used in the [tab_style()] function). If using [opt_table_font()] (which also
+#' has a `font` argument) we probably don't need to specify this vector of fonts
+#' since it is handled by its `add` option (which is `TRUE` by default).
+#'
+#'
+#' @examples
+#' # Use `exibble` to create a gt table;
+#' # attempting to modify the fonts used
+#' # for the `time` column is much safer
+#' # if `default_fonts()` is appended to
+#' # the end of the `font` listing in the
+#' # `cell_text()` call (the "Comic Sansa"
+#' # and "Menloa" fonts don't exist, but,
+#' # we'll get the first available font
+#' # from the `default_fonts()` set)
+#' tab_1 <-
+#'   exibble %>%
+#'   dplyr::select(char, time) %>%
+#'   gt() %>%
+#'   tab_style(
+#'     style = cell_text(
+#'       font = c(
+#'         "Comic Sansa", "Menloa",
+#'         default_fonts()
+#'       )
+#'     ),
+#'     locations = cells_body(columns = vars(time))
+#'   )
+#'
+#' @family Helper Functions
+#'
+#' @section Function ID:
+#' 7-22
+#'
+#' @export
+default_fonts <- function() {
+  c(
+    "-apple-system", "BlinkMacSystemFont", "Segoe UI", "Roboto",
+    "Oxygen", "Ubuntu", "Cantarell", "Helvetica Neue", "Fira Sans",
+    "Droid Sans", "Arial", "sans-serif"
+  )
+}
+
+#' Helper function for specifying a font from the Google Fonts service
+#'
+#' The `google_fonts()` helper function can be used wherever a font name should
+#' be specified. There are two instances where this helper can be used: the
+#' `name` argument in [opt_table_font()] (for setting a table font) and in that
+#' of [cell_text()] (used with [tab_style()]).
+#'
+#' @param name The complete name of a font available in Google Fonts.
+#'
+#' @return An object of class `google_fonts`.
+#'
+#' @family Helper Functions
+#' @section Function ID:
+#' 7-23
+#'
+#' @export
+google_fonts <- function(name) {
```

Comment:
because this is a single name, I am leaning towards `google_font(name)` (with no `s`)

Reply:
Good idea, I'll make that change.

Reply:
Okay, change is made. Great suggestion.